scanner.mll

```diff
@@ -4,37 +4,65 @@
 
 rule token = parse
   [' ' '\t' '\r' '\n'] { token lexbuf } (* Whitespace *)
-| "/*"     { comment lexbuf }           (* Comments *)
-| '('      { LPAREN }
-| ')'      { RPAREN }
-| '{'      { LBRACE }
-| '}'      { RBRACE }
-| ';'      { SEMI }
-| ','      { COMMA }
-| '+'      { PLUS }
-| '-'      { MINUS }
-| '*'      { TIMES }
-| '/'      { DIVIDE }
-| '='      { ASSIGN }
-| "=="     { EQ }
-| "!="     { NEQ }
-| '<'      { LT }
-| "<="     { LEQ }
-| ">"      { GT }
-| ">="     { GEQ }
-| "&&"     { AND }
-| "||"     { OR }
-| "!"      { NOT }
-| "if"     { IF }
-| "else"   { ELSE }
-| "for"    { FOR }
-| "while"  { WHILE }
-| "return" { RETURN }
-| "int"    { INT }
-| "bool"   { BOOL }
-| "void"   { VOID }
-| "true"   { TRUE }
-| "false"  { FALSE }
+| "/*"		{ comment lexbuf }           (* Comments *)
+| '('		{ LPAREN }
+| ')'		{ RPAREN }
+| '{'		{ LBRACE }
+| '}'		{ RBRACE }
+| '['		{ LBRACK }
+| ']'		{ RBRACK }
+| ';'		{ SEMI }
+| ','		{ COMMA }
+| '+'		{ PLUS }
+| '-'		{ MINUS }
+| '*'		{ TIMES }
+| '/'		{ DIVIDE }
+| '%'		{ MOD }
+| '=' 		{ ASSIGN }
+| "=="		{ REFEQ }
+| "==="		{ VEQ }
+| "!="		{ NEQ }
+| "!=="		{ VNEQ }
+| '<'		{ LT }
+| "<="		{ LEQ }
+| '>'		{ GT }
+| ">="		{ GEQ }
+| "and"		{ AND }
+| "or"		{ OR }
+| "not"		{ NOT }
+| "if"		{ IF }
+| "else"	{ ELSE }
+| "elseif"	{ ELSEIF }
+| "break"	{ BREAK }
+| "continue"	{ CONTINUE }
+| "for"		{ FOR }
+| "while"	{ WHILE }
+| "return"	{ RETURN }
+| "int"		{ INT }
+| "bool" 	{ BOOL }
+| "void"	{ VOID }
+| "string"	{ STRING }
+| "char"	{ CHAR }
+| "float"	{ FLOAT }
+| "true"	{ TRUE }
+| "false"	{ FALSE }
+| "->"		{ ARROW }
+| "@"		{ CONCAT }
+| "~" 		{ DEL }
+| "."		{ DECIMAL }
```

Comment:
Can we just call this "DOT" or something since it could be a function call/field access